etna/experimental/classification/classification.py

```diff
@@ -0,0 +1,128 @@
+from itertools import compress
+from typing import List
+from typing import Optional
+from typing import Set
+
+import numpy as np
+from sklearn.base import ClassifierMixin
+from sklearn.metrics import roc_auc_score
+
+from etna.experimental.classification.base import PickleSerializable
+from etna.experimental.classification.feature_extraction.base import BaseTimeSeriesFeatureExtractor
+from etna.loggers import tslogger
+
+
+class TimeSeriesBinaryClassifier(PickleSerializable):
+    """Class for holding time series binary classification."""
+
+    def __init__(
+        self, feature_extractor: BaseTimeSeriesFeatureExtractor, classifier: ClassifierMixin, threshold: float = 0.5
+    ):
+        """Init TimeSeriesClassifier with given parameters.
+
+        Parameters
+        ----------
+        feature_extractor:
+            Instance of time series feature extractor.
+        classifier:
+            Instance of classifier with sklearn interface.
+        threshold:
+            Positive class probability threshold.
+        """
+        self.feature_extractor = feature_extractor
+        self.classifier = classifier
+        self.threshold = threshold
+        self._classes: Optional[Set[int]] = None
+
+    def fit(self, x: List[np.ndarray], y: np.ndarray) -> "TimeSeriesBinaryClassifier":
+        """Fit the classifier.
+
+        Parameters
+        ----------
+        x:
+            Array with time series.
+        y:
+            Array of class labels.
+
+        Returns
+        -------
+        :
+            Fitted instance of classifier.
+        """
+        self._classes = set(map(int, y))
+        x_tr = self.feature_extractor.fit_transform(x, y)
+        self.classifier.fit(x_tr, y)
+        return self
+
+    def predict(self, x: List[np.ndarray]) -> np.ndarray:
+        """Predict classes with threshold.
+
+        Parameters
+        ----------
+        x:
+           Array with time series.
+
+        Returns
+        -------
+        :
+            Array with predicted labels.
+        """
+        y_prob_pred = self.predict_proba(x)
+        y_pred = (y_prob_pred > self.threshold).astype(int)
+        return y_pred
+
+    def predict_proba(self, x: List[np.ndarray]) -> np.ndarray:
+        """Predict probabilities of the positive class.
+
+        Parameters
+        ----------
+        x:
+            Array with time series.
+
+        Returns
+        -------
+        :
+            Probabilities for classes.
+        """
+        if self._classes is None:
+            raise ValueError("Classifier is not fitted!")
+
+        x_tr = self.feature_extractor.transform(x)
+        y_probs = self.classifier.predict_proba(x_tr)[:, 0]
+        if 0 in self._classes:
```

Comment:
Why do we need this?

Reply:
Method return the probability of the positive class, so if there was no positive class samples in train part there will be only one column with the corresponding probabilities(for positive or for negative

Reply:
Seems like all the predict_proba logic has possible problem with class shufling.

May be we should proxy the behaviour of sklearn without additional slicing.

class_0 < class_1 or class_1 < class_2 and return 2d array

Reply:
It is 1d array in case of single class on train, that is why i do this tric

Reply:
Ok, but my point is to leave the behaviour of sklearn.
`self.classifier.predict_proba(x_tr)[:, 0]` -> it's not the sklearn api.

Reply:
Add restrictions for the classes